index.js

```diff
@@ -1134,20 +1028,19 @@ module.exports = function (log, indexesPath) {
         )
       }),
 
-      // create missing indexes, if any
+      // update all relevant indexes, creating missing indexes, if any
       //
       // this needs to happen after loading lazy indexes because some
       // lazy indexes may have failed to load, and are now considered missing
       push.asyncMap((_, next) => {
-        const opsMissingIdx = detectOpsMissingIndexes(operation)
-        if (opsMissingIdx.length === 0) return next()
-        debug('missing indexes: %o', opsMissingIdx)
-        createIndexes(opsMissingIdx, next)
+        const ops = []
+        ops.push({ data: { indexName: 'seq' } }) // always ensure seq updated
+        forEachLeafOperationIn(operation, (op) => {
+          ops.push(op)
+        })
+        updateIndexes(ops, next)
       }),
 
-      // ensure that the seq->offset index is synchronized with the log
-      push.asyncMap((_, next) => ensureSeqIndexSync(next)),
```

Comment:
this ensureSeqIndexSync is now only used in 1 place. Can we in that place just call  ensureIndexSync({ data: { indexName: 'seq' } }, .... and then remove that function?

Reply:
I'm now on mobile, but yes I'll do that later

Reply:
done